### ISSUE TITLE
Cannot install MicroG on ZUI (Android 6.0.1)

### ISSUE THREAD

**Issue by sorcer1122** [microg/android_packages_apps_GmsCore#254]
Hello everyone,

Would appreciate any help, have been fighting with this issue for a few days already.

Tried to install latest MicroG on Zuk Z2 Pro, it has a custom version of Android - zui, made by Lenovo. First, I surprisingly found some Google apps there (ROM should come Google free since it is chinese), so I deleted three main components from /system/priv-app - gmscore, GoogleServicesFramework and GoogleLoginService. After that I was able to install MicroG gmscore but I am not able to run it. There is no icon on the desktop. In the settings I can see it is installed and I can uninstall it from there.

I also tried to install MicroG via recovery using Aroma installer but still no luck.

ROM obviously does not have signature spoofing but I guess it should not be a problem since on all previous ROMs I was able to launch MicroG even without signature spoofing.

It seems like zui blocks it somehow from creating an icon. I checked SafeCenter, nothing there blocking it.

Any ideas how to fix it?

**Comment by ale5000-git** [comment]
@sorcer1122: microG cannot be run, the only icon present is the one to change settings.
I don't know for sure but maybe the icon to change settings is in a different position on your phone, sometimes the icon is here, just harder to find.

**Comment by sorcer1122** [comment]
@ale5000-git yes,yes, this is what I meant - there is not icon 'MicroG Settings'. No, it is definitely not there, I searched every single corner. Also, on zukfans.eu website some people also had the same problem, so I guess it is zui's issue. I am just trying to understand how to fix it.

**Comment by ale5000-git** [comment]
@sorcer1122: Try my [flashable ZIP](https://forum.xda-developers.com/showthread.php?t=3432360), it should automatically remove some GApps and install microG.

If you have used an Aroma installer before there may be additional traces left, I suggest to remove these before flashing.

**Comment by ale5000-git** [comment]
@sorcer1122: Any update on this?

**Comment by sorcer1122** [comment]
@ale5000-git nope, I stopped using ZUI.

**Comment by ale5000-git** [comment]
Just for reference I got an idea, maybe using an alternative launcher will fix the problem but I cannot be sure.

**Comment by sorcer1122** [comment]
@ale5000-git I tried it already, did not work :(

**Comment by juneyao** [comment]
The problem is that you've got a chink phone. Pervert greedy chinks don't let mainland inmate chinks to use foreign stuff.
Somehow you bought a phone with the chink version of the software. It won't let you install real Gapps or MicroG, or anything named com.google.android.gsm  that is not their spyware version of it.
However, no respectable app, such as Signal/Whatsapp, will run with the chink version of gapps that comes with a chink phone because the signature is different. There is a chink spyware instant messaging app, wechat, that can work with the chink version of gapps, but you should not be using that anyway.

Remember that chinks built the phone copying everything from foreigners. The phone is android based, and android is all about freedom for users. But the always pervert and greedy chinks of course will not respect that.

The best you can do is return the chink phone to however cheated you, get your money back and don't ever waste your money or deal with chink shit no more. If you can't, try to steal your money back from a chink on the street. Do not hesitate to use violence.

**Comment by Nanolx** [comment]
dafuq did I read?

**Comment by ale5000-git** [comment]
@sorcer1122: Can you please try [microG unofficial installer](https://forum.xda-developers.com/showthread.php?t=3432360) 1.0.18 alpha?
I have done a lot of changes that probably fix the problem but I would like to be sure.

**Comment by sorcer1122** [comment]
@ale5000-git Unfortunately, I am not using ZUI anymore, so I am not able to test it. Sorry.

**Comment by ale5000-git** [comment]
No problem, I hope someone else test it.
